cmd/frontend/graphqlbackend/search_results.go

```diff
@@ -560,7 +560,20 @@ func (r *searchResolver) evaluateLeaf(ctx context.Context) (*SearchResultsResolv
 	// If the request specifies stable:truthy, use pagination to return a stable ordering.
 	if r.query.BoolValue("stable") {
 		result, err := r.paginatedResults(ctx)
-		if err == nil && !result.cursor.Finished {
+		if err != nil {
+			return nil, err
+		}
+		if result == nil {
+			// paginatedResults should be ensuring that result is
+			// non-nil, but the possibility that this changes at a
+			// later stage is too scary to entertain.
+			return &SearchResultsResolver{}, nil
```

Comment:
this could be really hard to debug as-is ("I get no errors but no results are returned!"), I would panic instead
```suggestion
			panic("stable search: paginated search returned nil results")
```

Reply:
Am fine with that!